hir2/src/ir/context.rs

```diff
@@ -276,6 +276,35 @@ impl Context {
         }
     }
 
+    /// Changes the destination of the `branch_inst` instruction to `new_dest` if the current
+    /// destination is `old_dest`.
+    ///
+    /// NOTE: Panics if `branch_inst` is not a branch instruction.
+    pub fn change_branch_destination(
+        &self,
+        mut branch_inst: OperationRef,
+        old_dest: BlockRef,
+        new_dest: BlockRef,
+    ) {
+        let mut borrow = branch_inst.borrow_mut();
+        let op = borrow.as_mut().as_operation_mut();
+        assert!(
+            op.as_trait::<dyn BranchOpInterface>().is_some(),
+            "expected branch instruction, got {branch_inst:?}"
+        );
+
+        let dest_operand_groups: Vec<usize> = op
```

Comment:
*Nit:* In general I would like to avoid heap allocations as much as possible in the core of the compiler. In pretty much every situation where you'd reach for `Vec`, `SmallVec` can be substituted, with the choice of size picked based on what is likely to be sufficient for most scenarios (i.e. it isn't important to make the `SmallVec` large enough to handle all inputs on the stack, just avoid allocations on average). Similarly, `String` can be replaced with `CompactString` for the same sorts of uses (for short strings anyway, for anything longish, just use `String`).

Reply:
Got it. Will do.